src/GraphQL.Tests/Instrumentation/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,245 @@
+#if NET5_0_OR_GREATER
+
+using System.Diagnostics;
+using GraphQL.Instrumentation;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace GraphQL.Tests.Instrumentation;
+
+public class OpenTelemetryTests : IDisposable
+{
+    private readonly List<Activity> _exportedActivities = new();
+    private readonly IHost _host;
+    private readonly IDocumentExecuter<ISchema> _executer;
+    private readonly IGraphQLTextSerializer _serializer;
+
+    public OpenTelemetryTests()
+    {
+        // configure services
+        _host = new HostBuilder()
+            .ConfigureServices(services =>
+            {
+                services.AddOpenTelemetry()
+                    .WithTracing(b => b
+                        .AddSource(TelemetryProvider.SourceName) // need to specify the source name to be traced
```

Comment:
Assuming that all users will need the source name constant, should we rename the class to `GraphQLTelemetryProvider` or something?  `TelemetryProvider` seems pretty generic.

Reply:
Sounds good

Reply:
Agree

Reply:
done; please review.